src/main/java/technology/dice/dicefairlink/config/ReplicasDiscoveryMode.java

```diff
@@ -0,0 +1,15 @@
+package technology.dice.dicefairlink.config;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum ReplicasDiscoveryMode {
+  AWS_API,
+  SQL_MYSQL;
+
+  public static Optional<ReplicasDiscoveryMode> fromStringInsensitive(String candidate) {
+    return Arrays.stream(ReplicasDiscoveryMode.values())
+        .filter(mode -> mode.toString().equalsIgnoreCase(candidate))
+        .findAny();
```

Comment:
`return Optional.of(ReplicasDiscoveryMode.valueOf(candidate.toUpperCase()));` ?

Reply:
That has a different semantic: throws if not found, so the Optional would never be empty,defeating the purpose

Reply:
But map-based `.valueOf` is so much faster than string-to-string comparison. Do we really want to allow something unexpected to be allowed as inputr here?